### PR TITLE
Chaudhryma

### PR DESCRIPTION
**Pull request by Mariachaudhry**

### CODE REVIEW

**Review by puzikovam2 — CHANGES_REQUESTED** [review]

**Review by andreyeve — CHANGES_REQUESTED** [review]

File: src/test/java/ServiceNow/GCP/StepsImplementation/GCPNotifications_NativeViewStepImpl.java
[the remark this answers is not in the document]
Comment (Mariachaudhry) [reply src/test/java/ServiceNow/GCP/StepsImplementation/GCPNotifications_NativeViewStepImpl.java:1]: deeted 5 sec sleep and "System Notification" is part of assertequalswithmessage which is recommended by Diego.

File: src/test/java/ServiceNow/NERD/StepsImplementation/NativeViewImpersonateUser.java
[the remark this answers is not in the document]
Comment (Mariachaudhry) [reply src/test/java/ServiceNow/NERD/StepsImplementation/NativeViewImpersonateUser.java:1]: This is for impersonation and for NERD application. screenshot were not working for me and Diego is looking into this issue which is why these 2 lines are commented to uncomment once the issue is resolved.

File: src/test/java/ServiceNow/ETracking/StepsImplementation/EtrackAssetsRecords_NativeViewStepImplimentation.java
[the remark this answers is not in the document]
Comment (Mariachaudhry) [reply src/test/java/ServiceNow/ETracking/StepsImplementation/EtrackAssetsRecords_NativeViewStepImplimentation.java:1]: These are needed to use specific dropdown values and send keys method.

File: src/test/java/ServiceNow/ETracking/StepsImplementation/EtrackAssetsRecords_NativeViewStepImplimentation.java [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    public static void thereIsAFieldCalled(String decalNumber) {
+        CommonUtils.waitForVisibility(etrackAssetsRecords_NativeViewPage.nVEtrackAssetsNewRecordDecalNumberLabel);
+        CommonUtils.assertEqualsWithMessage(etrackAssetsRecords_NativeViewPage.nVEtrackAssetsNewRecordDecalNumberLabel.getText(), decalNumber, "--Verify the Decal number field--");
+    }
+
+    public static void thereIsNoFieldCalled(String barcode) {
+        CommonUtils.assertTrue(!barcode.equals(etrackAssetsRecords_NativeViewPage.nVEtrackAssetsNewRecordDecalNumberLabel.getText()));
+    }
+}
+
```
[the remark this answers is not in the document]
Comment (Mariachaudhry) [reply src/test/java/ServiceNow/ETracking/StepsImplementation/EtrackAssetsRecords_NativeViewStepImplimentation.java:183]: Resolved

**Review by andreyeve — APPROVED** [review]

**Review by puzikovam2 — APPROVED** [review]
Reviewed and approved puzikovam2

File: src/test/java/ServiceNow/NERD/StepsImplementation/NativeViewImpersonateUser.java [hunk 2]
```diff
@@ -94,4 +96,40 @@ public void impersonateOGCRUser() throws TestingException {
...
+        MiscUtils.sleep(2000);
+        nativeViewDashPage.clickImpersonateUserLink();
+        MiscUtils.sleep(2000);
+        nativeViewDashPage.clickImpersonateSearchDD();
+        MiscUtils.sleep(3000);
+        nativeViewDashPage.enterTextImpersntSearchBox(EtrackAssetsRecords_NativeView_Constants.ETRACKING_USER_NAME);
+        MiscUtils.sleep(5000);
+        nativeViewImpersonateUserPage.nativeViewLink.click();
+        MiscUtils.sleep(2000);
+//        CucumberLogUtils.takeScreenShot(HooksSteps.scenario);
```
Comment (zamant2) [inline src/test/java/ServiceNow/NERD/StepsImplementation/NativeViewImpersonateUser.java:115]: please remove commented line#115 and #133

**Review by georgeb79 — APPROVED** [review]

**Review by zamant2 — APPROVED** [review]

**Pull request merged**